core/trino-main/src/main/java/io/trino/sql/planner/PartitioningScheme.java

```diff
@@ -77,6 +81,7 @@ public PartitioningScheme(
         checkArgument(!replicateNullsAndAny || columns.size() <= 1, "Must have at most one partitioning column when nullPartition is REPLICATE.");
         this.replicateNullsAndAny = replicateNullsAndAny;
         this.bucketToPartition = requireNonNull(bucketToPartition, "bucketToPartition is null");
+        this.hashPartitionCount = requireNonNull(hashPartitionCount, "hashPartitionCount is null");
```

Comment:
you could check state that partitoning is system handle when `hashPartitionCount` is present